interview-questions.md

```diff
@@ -9,41 +9,86 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is object-oriented programming? How is it different than functional programming?
 
 Your answer:
+Object oriented programming focuses around creating and manipulating objects. Functional programming focuses 
+on creating and utilizing functions
 
 Researched answer:
+Object oriented programming is a model of computer programming that focuses on creating and manipulating
+objects. Functional programming is another model of computer programming that focuses on functions to create
+software.
+
+https://www.codingdojo.com/blog/what-is-functional-programming#:~:text=Functional%20programming%20(FP)%20is%20an,by%20applying%20and%20composing%20functions.
+https://www.techtarget.com/searchapparchitecture/definition/object-oriented-programming-OOP#:~:text=Object%2Doriented%20programming%20(OOP)%20is%20a%20computer%20programming%20model,has%20unique%20attributes%20and%20behavior.
 
 1. What is the difference between a Float and an Integer in Ruby?
 
 Your answer:
+In Ruby, Integer is a whole number and a Float has a decimal value.
 
 Researched answer:
+In Ruby, an Integer is defined as a number without a decimal (example: 5) and a Float is
+a number with a decimal (example: 5.1 -- example2: 5.0).
+
+https://www.digitalocean.com/community/tutorials/understanding-data-types-in-ruby#:~:text=Ruby%20will%20consider%20any%20number,float%20(as%20in%20138.0%20).
 
 1. Ruby has an implicit return. What does that mean?
 
 Your answer:
+An implicit return means you do not have to define what to return, the last line of code
+before the 'end' is returned.
 
 Researched answer:
+In Ruby, the implicit return is the last line of code inside the method before the 'end' keyword.
+You do not have to declare what is returned unlike in Javscript.
+
+https://www.learnhowtoprogram.com/ruby-and-rails/basic-ruby/defining-methods-in-ruby#:~:text=The%20implicit%20return%20of%20a,method%20is%20exactly%20the%20same.
 
 1. What is a block in Ruby?
 
 Your answer:
+A block in ruby is the code that runs.what is a block in ruby
+
 
 Researched answer:
+In Ruby, a block contains your code that was assigned with a name. The block can be invoked by
+calling its name.
+
+https://www.tutorialspoint.com/ruby/ruby_blocks.htm#:~:text=Similarly%2C%20Ruby%20has%20a%20concept,as%20that%20of%20the%20block.
 
```

Comment:
Try to include more examples or elaborate some more on these interview questions! This will ensure you convey the full extent of your knowledge to the interviewer.